src/pkt_handlers.c

```diff
@@ -2043,34 +2043,47 @@ void NF_ingress_vrf_name_handler(struct channels_list_entry *chptr, struct packe
 {
   struct pkt_vlen_hdr_primitives *pvlen = (struct pkt_vlen_hdr_primitives *) ((*data) + chptr->extras.off_pkt_vlen_hdr_primitives);
 
-  if (check_pipe_buffer_space(chptr, pvlen, MAX_VRF_NAME)) {
+  if (strlen(pptrs->ingress_vrf_name) == 0) {
+    return;
+  }
+
+  if (check_pipe_buffer_space(chptr, pvlen, PmLabelTSz+strlen(pptrs->ingress_vrf_name)+1)) {
```

Comment:
I didn't see it before; `strlen()` is expensive. 

Might have been a good idea to define:

```
int ingress_vrf_name_len = strlen(pptrs->ingress_vrf_name) +1;
```

Change `if()` clause to ` == 1` and use the stack var everywhere else.
And change the 

And immed